webapp/src/components/backstage/category_selector.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import {SelectComponentsConfig, components as defaultComponents} from 'react-select';
+import {useSelector} from 'react-redux';
+import {makeGetCategoriesForTeam} from 'mattermost-redux/selectors/entities/channel_categories';
+
+import {ChannelCategory} from 'mattermost-redux/types/channel_categories';
+import {GlobalState} from 'mattermost-redux/types/store';
+
+import {StyledCreatable} from './styles';
+
+export interface Props {
+    id?: string;
+    onCategorySelected: (categoryName: string) => void;
+    categoryName?: string;
+    isClearable?: boolean;
+    selectComponents?: SelectComponentsConfig<ChannelCategory>;
+    isDisabled: boolean;
+    captureMenuScroll: boolean;
+    shouldRenderValue: boolean;
+    placeholder?: string;
+}
+
+const getCategoriesForTeam = makeGetCategoriesForTeam();
+const getMyCategories = (state: GlobalState) => getCategoriesForTeam(state, state.entities.teams.currentTeamId);
+
+const CategorySelector = (props: Props & { className?: string }) => {
+    const selectableCategories = useSelector(getMyCategories);
+
+    const options = React.useMemo(() => {
+        return selectableCategories
+            .filter((category) => category.type !== 'direct_messages')
+            .map((category) => ({value: category.display_name, label: category.display_name}));
+    }, [selectableCategories]);
+
+    const onChange = (option: {label: string; value: string}, {action}: {action: string}) => {
+        if (action === 'clear') {
+            props.onCategorySelected('');
+        } else {
+            props.onCategorySelected(option.value);
+        }
+    };
+
+    const components = props.selectComponents || defaultComponents;
+
+    return (
+        <StyledCreatable
+            className={props.className}
+            id={props.id}
+            controlShouldRenderValue={props.shouldRenderValue}
+            options={options}
+            onChange={onChange}
+            defaultMenuIsOpen={false}
+            openMenuOnClick={true}
+            isClearable={props.isClearable}
+            value={{value: props.categoryName, label: props.categoryName}}
+            placeholder={props.placeholder || 'Add channel to category'}
```

Comment:
Since not that long ago, we can internationalize text in Playbooks, so this is a good opportunity to use it.

You simply just add `const {formatMessage} = useIntl();` at the beginning of the component (the import for this is `import {useIntl} from 'react-intl';`) and change this line to
```suggestion
            placeholder={props.placeholder || formatMessage({defaultMessage: 'Add channel to category'})}
```

Reply:
Also I had to remove `placeholder` from upper component `categorize_playbook_run` so it use `formatMessage` instead of un-translated `props.placeholder`